sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -33,18 +33,29 @@ object GpuHashJoin {
       rightKeys: Seq[Expression],
       condition: Option[Expression]): Unit = {
     val keyDataTypes = (leftKeys ++ rightKeys).map(_.dataType)
-    if (keyDataTypes.exists(dtype =>
-      dtype.isInstanceOf[ArrayType] || dtype.isInstanceOf[StructType]
-        || dtype.isInstanceOf[MapType])) {
-      meta.willNotWorkOnGpu("Nested types in join keys are not supported")
+    if (keyDataTypes.exists(dType =>
+      dType.isInstanceOf[ArrayType] || dType.isInstanceOf[MapType])) {
+      meta.willNotWorkOnGpu("ArrayType or MapType in join keys are not supported")
+    }
+
+    val unSupportNonEqualCondition = () => if (condition.isDefined) {
+      meta.willNotWorkOnGpu(s"$joinType joins currently do not support conditions")
+    }
+    val unSupportStructKeys = () => if (keyDataTypes.exists(_.isInstanceOf[StructType])) {
```

Comment:
nit:

I personally find it much more readable to do a def instead of a val for functions like this.

```
def unSupportNonEqualCondition:
  ...

def unSupportStructKeys:
  ...
```